cli/src/pcluster/models/cluster.py

```diff
@@ -355,7 +355,7 @@ def _upload_config(self):
             if self.config.source_config:
                 result = AWSApi.instance().s3.put_object(
                     bucket_name=self.bucket.name,
-                    body=yaml.dump(self.config.source_config),
+                    body=yaml.dump(ClusterSchema().dump(self.config)),
```

Comment:
I'd upload two different files, the original one and the dumped one (with defaults).

The first one is useful if we want to retrieve the original config written by the customer.

Reply:
Agreed, I will add a ToDo. I chose to upload one file to avoid change in cookbook

Reply:
Fixed